### ISSUE TITLE
'NSOpenGLContext' is deprecated

### ISSUE THREAD

**Issue by daemonchen** [EngoEngine/engo#686]
when I install  with `go get -u github.com/EngoEngine/engo`, got the warning like this:
```
warning: 'NSOpenGLContext' is deprecated: first deprecated in macOS 10.14 - Please use Metal or MetalKit
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSOpenGL.h:202:12: note: 'NSOpenGLContext' has been explicitly marked deprecated here
```

**Comment by Noofbiz** [comment]
Yeah, macOS is removing OpenGL sometime in the future. I'm actually working on moving the render system to Vulkan/Metal [here](https://github.com/Noofbiz/vulkanRenderSystem). Once that's up and running, I'm going to put it into common as the normal render system but with the build tag "vulkan" that you can use to build without OpenGL at all. I'll keep this issue open to track the progress on this, since I can't seem to find a similar issue ^_^
